docs/codegen/codegen.rst

```diff
@@ -0,0 +1,205 @@
+Introduction
+------------
+
+The `pydy.codegen` package contains various tools to generate numerical code
+from symbolic descriptions of the equations of motion of systems. It allows you
+to generate code using a variety of backends depending on your needs. The
+generated code can also be auto-wrapped for immediate use in a Python session
+or script. Each component of the code generators and wrappers are accessible so
+that you can use just the raw code or the wrapper versions.
+
+We currently support three backends:
+
+`lambdify`
+   This uses the `sympy.utilities.lambdify` module to generate NumPy aware
+   Python code which is defined in a Python `lambda` function.
+`Theano`
+   This uses the `sympy.printers.theano_code` module to generate Theano trees
+   that are compiled into low level code.
+`Cython`
+   The uses SymPy's C code printer utilities and Cython to wrap the code for
+   use in Python.
+
+Example Use
+-----------
+
+The simplest entry point to the code generation tools is through the `System`
+class.
+
+.. code:: pycon
+
+   >>> from pydy.models import multi_mass_spring_damper
+   >>> sys = multi_mass_spring_damper()
+   >>> type(sys)
+   <class 'pydy.system.System'>
+   >>> rhs = sys.generate_ode_function()
```

Comment:
Which generator is used here? Perhaps give the default in the list of backends above?